File: badcrypto/src/main/java/com/lab/BadCrypto.java
package com.lab;

import javax.crypto.Cipher;
import javax.crypto.KeyGenerator;
import javax.crypto.SecretKey;
import javax.crypto.spec.GCMParameterSpec;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;
import java.security.KeyPairGenerator;
import java.security.MessageDigest;
import java.security.SecureRandom;
import java.security.Signature;

/** Deliberately mixed crypto so the CBOM scan has something to find. */
public class BadCrypto {

    // --- findings the scan should raise ---

    public byte[] legacyHash(byte[] in) throws Exception {
        MessageDigest md = MessageDigest.getInstance("MD5");      // broken
        return md.digest(in);
    }

    public byte[] legacyHash2(byte[] in) throws Exception {
        MessageDigest md = MessageDigest.getInstance("SHA-1");    // deprecated
        return md.digest(in);
    }

    public byte[] desEncrypt(byte[] data) throws Exception {
        KeyGenerator kg = KeyGenerator.getInstance("DES");        // 56-bit
        SecretKey k = kg.generateKey();
        Cipher c = Cipher.getInstance("DES/ECB/PKCS5Padding");    // ECB too
        c.init(Cipher.ENCRYPT_MODE, k);
        return c.doFinal(data);
    }

    public byte[] hardcodedIv(byte[] data) throws Exception {
        byte[] key = "0123456789abcdef".getBytes();               // hardcoded key
        byte[] iv  = new byte[16];                                // static IV
        Cipher c = Cipher.getInstance("AES/CBC/PKCS5Padding");
        c.init(Cipher.ENCRYPT_MODE, new SecretKeySpec(key, "AES"), new IvParameterSpec(iv));
        return c.doFinal(data);
    }

    public void weakRsa() throws Exception {
        KeyPairGenerator kpg = KeyPairGenerator.getInstance("RSA");
        kpg.initialize(1024);                                     // Shor-broken and weak today
        kpg.generateKeyPair();
    }

    public void rsa2048() throws Exception {
        KeyPairGenerator kpg = KeyPairGenerator.getInstance("RSA");
        kpg.initialize(2048);                                     // Shor-broken
        kpg.generateKeyPair();
    }

    public void ecdsaSign(byte[] data) throws Exception {
        KeyPairGenerator kpg = KeyPairGenerator.getInstance("EC");
        kpg.initialize(256);                                      // Shor-broken
        Signature s = Signature.getInstance("SHA256withECDSA");
        s.initSign(kpg.generateKeyPair().getPrivate());
        s.update(data);
        s.sign();
    }

    // --- the one correct path, for contrast ---

    public byte[] aesGcm(byte[] data) throws Exception {
        KeyGenerator kg = KeyGenerator.getInstance("AES");
        kg.init(256);                                             // Grover-weakened only
        byte[] nonce = new byte[12];
        SecureRandom.getInstanceStrong().nextBytes(nonce);
        Cipher c = Cipher.getInstance("AES/GCM/NoPadding");
        c.init(Cipher.ENCRYPT_MODE, kg.generateKey(), new GCMParameterSpec(128, nonce));
        return c.doFinal(data);
    }
}

File: badcrypto/py/bad_crypto.py
"""Deliberately mixed crypto for the CBOM scan (pyca/cryptography)."""
import hashlib
import os

from cryptography.hazmat.primitives import hashes, serialization
from cryptography.hazmat.primitives.asymmetric import rsa, ec, padding
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
from cryptography.hazmat.primitives.ciphers.aead import AESGCM


# --- findings the scan should raise ---

def md5_digest(data: bytes) -> bytes:
    return hashlib.md5(data).digest()                     # broken


def sha1_digest(data: bytes) -> bytes:
    h = hashes.Hash(hashes.SHA1())                        # deprecated
    h.update(data)
    return h.finalize()


def tripledes(data: bytes) -> bytes:
    key = b"0123456789abcdef01234567"                     # hardcoded
    c = Cipher(algorithms.TripleDES(key), modes.ECB())    # 3DES + ECB
    e = c.encryptor()
    return e.update(data) + e.finalize()


def weak_rsa():
    return rsa.generate_private_key(public_exponent=65537, key_size=1024)


def rsa_2048_encrypt(data: bytes) -> bytes:
    k = rsa.generate_private_key(public_exponent=65537, key_size=2048)
    return k.public_key().encrypt(
        data, padding.OAEP(padding.MGF1(hashes.SHA256()), hashes.SHA256(), None)
    )


def ecdsa_sign(data: bytes) -> bytes:
    k = ec.generate_private_key(ec.SECP256R1())           # Shor-broken
    return k.sign(data, ec.ECDSA(hashes.SHA256()))


# --- the one correct path, for contrast ---

def aes_gcm(data: bytes) -> bytes:
    key = AESGCM.generate_key(bit_length=256)             # Grover-weakened only
    nonce = os.urandom(12)
    return nonce + AESGCM(key).encrypt(nonce, data, None)
